pkg/handlers/generic/lifecycle/csi/nutanix-csi/handler.go

```diff
@@ -215,11 +204,16 @@ func (n *NutanixCSI) handleHelmAddonApply(
 		},
 	}
 
-	if err = client.ServerSideApply(ctx, n.client, snapshotChart, client.ForceOwnership); err != nil {
-		return fmt.Errorf(
-			"failed to apply nutanix-csi-snapshot installation HelmChartProxy: %w",
-			err,
-		)
+	// We use a slice of pointers to satisfy the gocritic linter rangeValCopy check.
+	for _, cp := range []*caaphv1.HelmChartProxy{storageChartProxy, snapshotChartProxy} {
+		if err = controllerutil.SetOwnerReference(&req.Cluster, cp, n.client.Scheme()); err != nil {
```

Comment:
i had also fixed this thru https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/523

Reply:
Thanks for the heads-up. I'll rebase.